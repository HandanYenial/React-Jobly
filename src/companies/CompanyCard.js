import React from "react";
import { Link } from "react";
import "./CompanyCard.css";

/** 1. Show information about a company
 *  2. It is rendered by CompanyList to show a "card" for each company
 */

function CompanyCard({ name, description, logoUrl, handle }){
    console.debug("CompanyCard" , "logoUrl");

    return(
        <Link className="CompanyCard card" to={`/companies/${handle}`}>
            <div className = "class-body">
                <h6 className = "card-title">
                    {name}
                    {logoUrl && <img src={logoUrl} alt={name} className="float-right ml-5" />}
                </h6>
                <p><small>{description}</small></p>
            </div>
        </Link>
    );
}

export default CompanyCard;