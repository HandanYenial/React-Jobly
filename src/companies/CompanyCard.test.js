import React from"react";
import CompanyCard from "./CompanyCard";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";


//snapshot test
it("matches snapshot with logo" , function(){
    const { asFragment } = render(
    <MemoryRouter>
        <CompanyCard
             handle = "test"
             name = "Test Company"
             description = "Testing Company Card"
             logo_url = "https://logos.flamingtext.com/Word-Logos/test-design-sketch-name.png"
        />
    </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot without logo" , function(){
    const { asFragment } = render(
        <MemoryRouter>
            <CompanyCard
                handle = "testwithoutlogo"
                name = "Test Company without logo"
                description = "Testing Company Card without logo"
            />

        </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});