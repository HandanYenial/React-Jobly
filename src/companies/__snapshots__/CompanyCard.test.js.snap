// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with logo 1`] = `
<DocumentFragment>
  <a
    class="CompanyCard card"
    href="/companies/test"
  >
    <div
      class="card-body"
    >
      <h6
        class="card-title"
      >
        Test Company
      </h6>
      <p>
        <small>
          Testing Company Card
        </small>
      </p>
    </div>
  </a>
</DocumentFragment>
`;

exports[`matches snapshot without logo 1`] = `
<DocumentFragment>
  <a
    class="CompanyCard card"
    href="/companies/testwithoutlogo"
  >
    <div
      class="card-body"
    >
      <h6
        class="card-title"
      >
        Test Company without logo
      </h6>
      <p>
        <small>
          Testing Company Card without logo
        </small>
      </p>
    </div>
  </a>
</DocumentFragment>
`;
